vocode/streaming/agent/state_agent.py

```diff
@@ -427,7 +445,15 @@ async def handle_state(self, state_id_or_label: str):
             return out
 
         if state["type"] == "action":
-            return await self.compose_action(state)
+            try:
+                return await self.compose_action(state)
+            except Exception as e:
+                # invariant violation, not action error, because compose_action is supposed to do it's own error handling
+                self.json_transcript.entries.append(
+                    StateAgentTranscriptInvariantViolation(
+                        message=f"uncaught exception in compose_action. State is {state}"
+                    )
+                )
```

Comment:
what should we actually do here? As written, this code will pretty much just end the call (I'm EXPECTING that we'll only hit this in production but that's probably not true)